### PR TITLE
Refactor ResolveLongIdentAsModuleOrNamespace

### PR DESCRIPTION
**Pull request by kerams**
Fixed the names of a couple of values as well as removed an unnecessary list creation.

The `(atMostOne: ResultCollectionSettings)` parameter to the function is currently not used for anything, so I could remove it if you want.

### CODE REVIEW

File: src/Compiler/Checking/NameResolution.fs [hunk 1]
```diff
@@ -2385,7 +2385,7 @@ let rec ResolveLongIdentAsModuleOrNamespace sink (atMostOne: ResultCollectionSet
             UndefinedName(depth, error, id, suggestNames)
 
         let moduleOrNamespaces = nenv.ModulesAndNamespaces fullyQualified
-        let namespaceNotFound =
+        let namespaceOrModuleNotFound =
```
Comment (kerams) [inline src/Compiler/Checking/NameResolution.fs:2388]: This matches the error message below.

File: src/Compiler/Checking/NameResolution.fs [hunk 2]
```diff
@@ -2394,66 +2394,53 @@ let rec ResolveLongIdentAsModuleOrNamespace sink (atMostOne: ResultCollectionSet
 
         // Avoid generating the same error and name suggestion thunk twice It's not clear this is necessary
         // since it's just saving an allocation.
-        let mutable moduleNotFoundErrorCache = None
-        let moduleNotFound (modref: ModuleOrNamespaceRef) (mty: ModuleOrNamespaceType) (id: Ident) depth =
-            match moduleNotFoundErrorCache with
+        let mutable namespaceNotFoundErrorCache = None
+        let namespaceNotFound (modref: ModuleOrNamespaceRef) (mty: ModuleOrNamespaceType) (id: Ident) depth =
+            match namespaceNotFoundErrorCache with
```
Comment (kerams) [inline src/Compiler/Checking/NameResolution.fs:2399]: This matches the error message below.

**Review by psfinaki — APPROVED** [review]
Thanks for this @kerams.
As for removing the parameter - is there any potential backward compat issue @vzarytovskii?

**Comment by vzarytovskii** [comment]
It's internal, so we should be fine with removing it

**Review by T-Gro — APPROVED** [review]

**Comment by psfinaki** [comment]
@kerams feel free to resolve the comments and merge.

**Pull request merged**